tbx/static_src/javascript/components/mode-switcher.js

```diff
@@ -0,0 +1,61 @@
+import Cookies from 'js-cookie';
+
+const { ALLOWED_MODES } = window.GLOBALS;
+
+class ModeSwitcher {
+    static selector() {
+        return '[data-mode-switcher]';
+    }
+
+    constructor(node) {
+        this.node = node;
+        this.window = window;
+        this.html = document.querySelector('html');
+        this.modeRadios = node.querySelectorAll('[data-mode-switch]');
+
+        let { mode } = this.html.dataset;
+
+        // check for invalid mode values
+        if (!ALLOWED_MODES.includes(mode)) mode = '';
+
+        // if mode cookie is not set, use dark by default, otherwise use the cookie value
+        this.mode = mode === '' ? 'dark' : mode;
+
+        this.toggleMode();
+        this.bindEventListeners();
+    }
+
+    bindEventListeners() {
+        this.modeRadios.forEach((radio) => {
+            radio.addEventListener('change', () => {
+                if (radio.checked) {
+                    const mode = radio.value;
+                    if (mode === this.mode) return;
+                    this.mode = mode;
+                    this.toggleMode();
+                }
+            });
+        });
+
+        this.node.querySelectorAll('form').forEach((form) => {
+            form.addEventListener('submit', (e) => {
+                e.preventDefault();
+            });
+        });
+    }
+
+    toggleMode() {
+        ALLOWED_MODES.forEach((allowedMode) => {
+            if (allowedMode !== this.mode) {
+                this.html.classList.remove(`mode-${allowedMode}`);
+            }
+        });
+        this.html.classList.add(`mode-${this.mode}`);
+        this.html.dataset.mode = this.mode;
+        Cookies.set('mode', this.mode, {
```

Comment:
Suggestion: I wonder if we should call this something more specific - we use 'torchbox-cookie' ([ref](https://github.com/torchbox/torchbox.com/blob/main/tbx/static_src/javascript/components/cookie-message.js#L16)) for the cookie so perhaps something like 'torchbox-colour-mode' or 'torchbox-site-theme'?

Reply:
ah good thought - I had assumed it didn't matter too much as it will be scoped by domain, but I guess `torchbox-mode` is clearer. (theme is coral etc)

Reply:
Now updated